TEMP_CCAAS_TUTORIAL.md

```diff
@@ -0,0 +1,137 @@
+# Running Chaincode as Service with the Test Network
+
+The chaincode-as-a-service feature is a very useful and practical way to run 'Smart Contracts'. Traditionally the Fabric Peer has taken on the role of orchestrating the complete lifecycle of the chaincode. It required access to the Docker Daemon to create images, and start containers. Java, NodeJS and Go chaincode frameworks were explicitly known to the peer including how they should be built and started.
+
+As a result this makes it very hard to deploy into Kubernetes (K8S) style environments, or to run in any form of debug mode. Additionally, the code is being rebuilt by the peer therefore there is some degree of uncertainty about what dependencies have been pulled in.
+
+Chaincode-as-service requires you to orchestrate the build and deployment phase yourself. Whilst this is an additional step, it gives control back. The Peer still requires a 'chaincode package' to be installed. In this case this doesn't contain code, but the information about where the chaincode is hosted. (Hostname,Port,TLS config etc)
+
+## Fabric v2.4.1 Improvements
+
+We need to use the latest 2.4.1 release as this contains some improvements to make this process easier. The core functionality is available in earlier releases but requires more configuration.
+
+- The docker image for the peer contains a builder for chaincode-as-a-service preconfigured. This is named 'ccaasbuilder'. This removes the need to build your own external builder and repackage and configure the peer
+- The `ccaasbuilder` applications are included in the binary tgz archive download for use in other circumstances. The `sampleconfig/core.yaml` is updated as well to refer to 'ccaasbuilder'
+- The 2.4.1 Java Chaincode release has been updated to remove the need to write a custom bootstrap main class. (similar  to the NodeJS Chaincode. It is intended that this will be added to the go chaincode as well.
+
+## End-to-end with the the test-network
+
+The `test-network` and some of the chaincodes have been updated to support running chaincode-as-a-service. The commands below assume that you've got the latest fabric-samples cloned, along with the latest Fabric docker images. 
+
+It's useful to have two terminal windows open, one for starting the Fabric Network, and a second for monitoring all the docker containers.
+
+In your 'monitoring' window, run this to watch all activity from the all the docker containers on the `fabric_test` network
+
+```bash
+# from the fabric-samples repo
+./test-network/monitordocker.sh
+```
```

Comment:
You must start test-network before monitordocker.sh will succeed.
Also, cd to test-network first to be consistent with the below command.